### ISSUE TITLE
Shell recorder tests never fail / not executing properly

### ISSUE THREAD

**Issue by joni1993** [ElektraInitiative/libelektra#4702]
While extending some tests i discoverey that the tests `tests/shell_recorder/*.esr` never seem to fail.

In selftest.esr the mountpoint is `user:/tests`, then a command is issued `kdb set $Mountpoint/key value` and the check is `STDOUT: Create a new key user:/test/key with string "value"`.  (Notice test**s** vs test) - But during `make run_all` the tests don't seem to fail, while when executing the shell_recorder test manually they fail as expected.

As my new tests don't increase the coverage, but executing the tests manually does increase the coverage, i suspect the tests don't even get executed properly.

## Steps to Reproduce the Problem
- `tests/shell/shell_recorder/shell_recorder.sh ../tests/shell/shell_recorder/selftest.esr`

```=== FAILED stdout does not match expected pattern Create a new key user:/test/key with string "value"```

but
- `make run_all`
```...
Start 153: testshell_selftest
204/212 Test #153: testshell_selftest ...........................   Passed    0.25 sec
```

No matter how hard i try to make the tests fail during `make run_all` they always pass. But changing for example `plugins/hexnumber/README.md` to something that should fail makes them fail (as expected).

So this seems to be a problem with the files in ` tests/shell_recorder/` only.

## Expected Result
Tests should fail as they do on manual execution.

## Actual Result
Tests never fail - even if they do on manual execution

## System Information

- Elektra: master (4283744cc4689792a8143d55f5f49e6dc1eac81c)
- Operating System: Ubuntu 22.04

Sadly this is actually blocking me for [FLOSS T1].

**Comment by 0x6178656c** [comment]
Something in the CMake build chain is broken. 

In the generated file `build/tests/shell/shell_recorder/CTestTestfile.cmake` there is the line

```cmake
add_test(testshell_selftest "/workspace/build/tests/shell/shell_recorder/shell_recorder.sh" "/workspace/tests/shell/shell_recorder/run_all.sh")
```

But the file `/workspace/tests/shell/shell_recorder/run_all.sh` does not exist.

My PR  #4709 adds a check to the shellrecorder and it now fails if the file cannot be found.

**Comment by 0x6178656c** [comment]
As expected with the check whether the file exists we get several failing tests

```text
The following tests FAILED:
         14 - testtool_plugindatabase (Failed)
        149 - testshell_db_changes (Failed)
        150 - testshell_host (Failed)
        151 - testshell_listtest (Failed)
        152 - testshell_mathcheck (Failed)
        153 - testshell_profiletest (Failed)
        154 - testshell_script (Failed)
        155 - testshell_selftest (Failed)
```

**Comment by 0x6178656c** [comment]
Problem is now fixed in #4709.

Indeed, CMake created incorrect tests for the shell recorder. The filename was not set  correctly.

**Comment by 0x6178656c** [comment]
I disabled two tests in #4709: e6a6d00ce540f0a0e57fdbcb20045ff74604b8ad

In both cases there were some error messages because of unsupported cascading lookups. I think this is related to the new backend, but somebody more knowledgeable should have a look as well.
